code/citadel/hypospraymkii.dm

```diff
@@ -0,0 +1,215 @@
+#define HYPO_INJECT 1
+
+//A vial-loaded hypospray. Cartridge-based!
+/obj/item/reagent_containers/hypospray/mkii
+	name = "hypospray mk.II"
+	icon = 'icons/obj/citadel/hypospray.dmi'
+	icon_state = "hypo2"
+	var/list/allowed_containers = list(/obj/item/reagent_containers/glass/bottle/vial/small)
+	desc = "A new development from DeForest Medical, this new hypospray takes 30-unit vials as the drug supply for easy swapping."
+	volume = 0
+	amount_per_transfer_from_this = 5
+	possible_transfer_amounts = list(5,10,15)
+	var/mode = HYPO_INJECT
+	var/obj/item/reagent_containers/glass/bottle/vial/vial
+	var/loaded_vial = /obj/item/reagent_containers/glass/bottle/vial/small
+	var/spawnwithvial = TRUE
+	var/start_vial = null
+
+/obj/item/reagent_containers/hypospray/mkii/CMO
+	name = "hypospray mk.II deluxe"
+	allowed_containers = list(/obj/item/reagent_containers/glass/bottle/vial/small, /obj/item/reagent_containers/glass/bottle/vial/large)
+	icon_state = "cmo2"
+	ignore_flags = 1
+	desc = "The Chief Medical Officer's hypospray is identically functional to the base model, excepting that it can take larger vials in addition to regular sized. It is also able to penetrate harder materials and deliver more reagents per spray."
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | ACID_PROOF
+	loaded_vial = /obj/item/reagent_containers/glass/bottle/vial/large/preloaded/CMO
+	possible_transfer_amounts = list(5,10,15,30,60) //cmo hypo should be able to dump lots into it
+
+/obj/item/reagent_containers/hypospray/mkii/Initialize()
+	. = ..()
+	if(!spawnwithvial)
+		update_icon()
+		return
+	if (!start_vial)
+		start_vial = new loaded_vial(src)
+		vial = start_vial
+	update_icon()
+
+/obj/item/reagent_containers/hypospray/mkii/update_icon()
+	..()
+	icon_state = "[initial(icon_state)][vial ? "" : "-e"]"
+	if(ismob(loc))
+		var/mob/M = loc
+		M.update_inv_hands()
+	return
+
+/obj/item/reagent_containers/hypospray/mkii/examine(mob/user)
+	. = ..()
+	to_chat(user, "[src] is set to [mode ? "Inject" : "Spray"] contents on application.")
+
+/obj/item/reagent_containers/hypospray/mkii/proc/unload_hypo(obj/item/I, mob/user)
+	if((istype(I, /obj/item/reagent_containers/glass/bottle/vial)))
+		var/obj/item/reagent_containers/glass/bottle/vial/V = I
+		reagents.trans_to(V, reagents.total_volume)
+		reagents.maximum_volume = 0
+		V.forceMove(user.loc)
+		user.put_in_hands(V)
+		to_chat(user, "<span class='notice'>You remove the vial from the [src].</span>")
+		vial = null
+		update_icon()
+		playsound(loc, 'sound/weapons/empty.ogg', 50, 1)
+	else
+		to_chat(user, "<span class='notice'>This hypo isn't loaded!</span>")
+		return
+
+/obj/item/reagent_containers/hypospray/mkii/attackby(obj/item/I, mob/living/user)
+	if((istype(I, /obj/item/reagent_containers/glass/bottle/vial) && vial != null))
+		to_chat(user, "<span class='warning'>[src] can not hold more than one vial!</span>")
+		return FALSE
+	testing("we don't already have a vial in the hypo")
```

Comment:
Shouldn't you remove this

Reply:
pfff. Stray testing verbs. pffff.

Probably.